algorithms/rainbow/agent.py

```diff
@@ -0,0 +1,134 @@
+"""
+Wrapper including setup, training and evaluation functions for Rainbow DQN model
+"""
+import os
+import numpy as np
+import torch
+from torch import optim
+
+from algorithms.rainbow.model import RainbowDQN
+
+
+class Agent:
+    def __init__(self, args, env):
+        self.action_space = env.action_space
+        self.atoms = args.atoms
+        self.Vmin = args.V_min
+        self.Vmax = args.V_max
+        # Support (range) of z
+        self.support = torch.linspace(args.V_min, args.V_max, self.atoms).to(device=args.device)
+        self.delta_z = (args.V_max - args.V_min) / (self.atoms - 1)
+        self.batch_size = args.batch_size
+        self.n = args.multi_step
+        self.discount = args.discount
+
+        self.online_net = RainbowDQN(args, self.action_space).to(device=args.device)
+        if args.model and os.path.isfile(args.model):
+            # Always load tensors onto CPU by default, will shift to GPU if necessary
```

Comment:
Are you happy if I change the comment to this: ```
             Always load tensors onto CPU by default, will shift to GPU if necessary to avoid 
            GPU RAM surge when loading a model checkpoint as recommended in pytorch official 
            documentation.
            Source: https://pytorch.org/docs/stable/torch.html#torch.load ```

Reply:
wow this is all new to me. maybe instead of saying "if necessary" say that when you call load_state_dict it actually puts it onto the GPU at the end. It's only on the CPU at the start before fully going to GPU?